tests/std/tests/P0009R18_mdspan_extents/test.cpp

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <mdspan>
+#include <span>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+enum class IsNothrow : bool { no, yes };
+
+template <class Int, IsNothrow Nothrow = IsNothrow::yes>
+struct ConvertibleToInt {
+    constexpr operator Int() const noexcept(to_underlying(Nothrow)) {
+        return Int{1};
+    }
+};
+
+struct NonConvertibleToAnything {};
+
+template <class T>
+constexpr void check_implicit_conversion(T); // not defined
+
+template <class T, class... Args>
+concept NotImplicitlyConstructibleFrom =
+    constructible_from<T, Args...> && !requires(Args... args) { check_implicit_conversion<T>({args...}); };

```

Comment:
It might not make a difference for this test, but I think that this should either use `_STD declval<Args>()` or `_STD forward<Args>(args)`, to match `constructible_from` and avoid always asking the question for lvalues.

Reply:
I forgot we were in test code, thanks for omitting `_STD`. My insufficiently caffeinated brain... :scream_cat: